test/test_oauth2.py

```diff
@@ -0,0 +1,234 @@
+
+    assert response.status_code == 400
+    assert response.headers.get('Content-Type') == 'application/problem+json'
+    assert json.loads(response.data.decode('UTF-8')) == {'title': 'Invalid state (original)'}
+
+
+def test_it_redirects_when_checking(test_client):
+    response = test_client.get(
+        '/oauth2/check?' + urlencode({'code': 1234, 'state': json.dumps(
+            {'redirect_uri': 'http://www.example.com/client/redirect', 'client_id': 'client_id',
+             'original': ''})}))
+
+    assert response.status_code == 302
+    assert response.headers['Location'] == 'http://www.example.com/client/redirect?' + urlencode(
+        {'code': 1234, 'state': ''})
+
+
+def test_it_requires_client_id_when_exchanging(test_client):
+    response = test_client.post('/oauth2/token')
+
+    assert response.status_code == 400
+    assert response.headers.get('Content-Type') == 'application/problem+json'
+    assert json.loads(response.data.decode('UTF-8')) == {'title': 'Invalid client_id'}
+
+
+def test_it_requires_client_secret_when_exchanging(test_client):
+    response = test_client.post('/oauth2/token', data={'client_id': 'client_id'})
+
+    assert response.status_code == 400
+    assert response.headers.get('Content-Type') == 'application/problem+json'
+    assert json.loads(response.data.decode('UTF-8')) == {'title': 'Invalid client_secret'}
+
+
+def test_it_requires_redirect_uri_when_exchanging(test_client):
+    response = test_client.post('/oauth2/token',
+                                data={'client_id': 'client_id', 'client_secret': 'client_secret'})
+
+    assert response.status_code == 400
+    assert response.headers.get('Content-Type') == 'application/problem+json'
+    assert json.loads(response.data.decode('UTF-8')) == {'title': 'Invalid redirect_uri'}
+
+
+def test_it_requires_grant_type_when_exchanging(test_client):
+    response = test_client.post('/oauth2/token',
+                                data={'client_id': 'client_id', 'client_secret': 'client_secret',
+                                      'redirect_uri': 'http://www.example.com/client/redirect'})
+
+    assert response.status_code == 400
+    assert response.headers.get('Content-Type') == 'application/problem+json'
+    assert json.loads(response.data.decode('UTF-8')) == {'title': 'Invalid grant_type'}
+
+
+def test_it_requires_code_when_exchanging(test_client):
+    response = test_client.post('/oauth2/token',
+                                data={'client_id': 'client_id', 'client_secret': 'client_secret',
+                                      'redirect_uri': 'http://www.example.com/client/redirect',
+                                      'grant_type': 'authorization_code'})
+
+    assert response.status_code == 400
+    assert response.headers.get('Content-Type') == 'application/problem+json'
+    assert json.loads(response.data.decode('UTF-8')) == {'title': 'Invalid code'}
+
+
+@responses.activate
```

Comment:
Is there anyway to turn on Responses for _all_ tests? I don't want to accidentally make any real requests.